### DISCUSSION TITLE
Does RHF mutate the `values` prop passed to `useForm`?

### DISCUSSION THREAD

**Discussion by paulleonartcalvo** [react-hook-form/react-hook-form#10008] [Q&A]
There isnt much about this prop in the documentation, but I've encountered strange behavior with it recently. I currently have a functional react component which has a `currentSettings` prop which contains a valid set of form values. The `currentSettings` come from an object which is stored in a Redux Store. When i use this prop directly in the useForm hook like so: 

```
... useForm<FormValuesType>({values: currentSettings,...})
```

i observe that the `currentSettings` are actually mutated upstream in the react tree. However, when i create a structured clone like this:

```
const currentSettingsCopy = useMemo(() => structuredClone(currentSettings), [currentSettings])
const methods = useForm<FormValuesType>({values: currentSettingsCopy,...})
```

that behavior goes away, and the form behaves as expected. The way i expect the form to behave is that, when `currentSettings` changes, the form values change. But when the form values change, `currentSettings` does NOT change. Has anyone else observed similar behavior? Are there some common things i could be doing wrong?

**Comment by bluebill1049** [comment]
could you reproduce this in a codesandbox and rise an issue? we shouldn't do that.

**Reply by paulleonartcalvo** [reply]
Sure, ill try and spin one up tomorrow and share it here
